src/tracker/tracker_sampler.jl

```diff
@@ -0,0 +1,98 @@
+abstract type AbstractROI end
+mutable struct BoxROI{T <: AbstractArray, S <: AbstractArray} <: AbstractROI
+    img::T
+    bound::S
+end
+
+abstract type AbstractTrackerSampler end
+mutable struct CurrentSample{F <: Float64, S <: Symbol} <: AbstractTrackerSampler
+    overlap::F
+    search_factor::F
+    mode::S
+    sampling_ROI::SVector{4, Integer}
+    valid_ROI::SVector{4, Integer}
+    tracked_patch::SVector{4, Integer}
+    CurrentSample{F, S}(overlap, search_factor, mode) where{F <: Float64, S <: Symbol} = new(overlap, search_factor, mode)
+end
+
+CurrentSample(overlap::F = 0.99, search_factor::F = 1.8, mode::S = :positive) where{F <: Float64, S <: Symbol} =
+CurrentSample{F, S}(overlap, search_factor, mode)
+
+function sample_tracker(sampler::CurrentSample, box::BoxROI)
+    sampler.tracked_patch = box.bound
+    sampler.valid_ROI = SVector{4, Integer}([1, 1, size(box.img, 1), size(box.img, 2)])
+
+    height = box.bound[3] - box.bound[1] + 1
+    width = box.bound[4] - box.bound[2] + 1
+
+    ROI_min_y = max(0, floor(Integer, box.bound[1] - height*((sampler.search_factor - 1) / 2) + 1))
+    ROI_min_x = max(0, floor(Integer, box.bound[2] - width*((sampler.search_factor - 1) / 2) + 1))
+
+    ROI_height = min(floor(Integer, height*sampler.search_factor), size(box.img, 1) - ROI_min_y + 1)
+    ROI_width = min(floor(Integer, width*sampler.search_factor), size(box.img, 2) - ROI_min_x + 1)
+
+    ROI_max_y = ROI_min_y + ROI_height - 1
+    ROI_max_x = ROI_min_x + ROI_width - 1
+
+    ROI = SVector{4, Integer}([ROI_min_y, ROI_min_x, ROI_max_y, ROI_max_x])
+
+    sample_pos_neg_roi(sampler, box, ROI)
+end
+
+function sample_pos_neg_roi(sampler::CurrentSample, box::BoxROI, ROI::SVector{4, Integer})
+    println(ROI)
```

Comment:
use `@info` if you intend to output some information.

DO be aware that this algorithm might be run for many times in a script, and `sample_pos_neg_roi` will called run for many^2 times. Such information would be kind of junk in that case.

Reply:
My bad. I was using `println` for debugging. Missed it while cleaning. Thanks.